### PR TITLE
Update modules with nsp issues

### PR DESCRIPTION
**Pull request by vladikoff**
@rfk r? or anyone else?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,15 +12,12 @@
     "bewit"
   ],
   "engines": {
-    "node": ">=0.10.33"
+    "node": ">=6"
```
Comment (vladikoff) [inline package.json]: modules below are to keep compat with node 6

**Comment by vladikoff** [comment]
Without this patch: 

```
➜  hapi-auth-hawk git:(master) nsp check package.json      
(+) 3 vulnerabilities found
┌───────────────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│               │ Prototype pollution attack                                                                                             │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Name          │ hoek                                                                                                                   │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ CVSS          │ 4 (Medium)                                                                                                             │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Installed     │ 2.16.3                                                                                                                 │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Vulnerable    │ <= 4.2.0 || >= 5.0.0 < 5.0.3                                                                                           │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Patched       │ > 4.2.0 < 5.0.0 || >= 5.0.3                                                                                            │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Path          │ hapi-auth-hawk@3.0.1 > boom@2.10.1 > hoek@2.16.3                                                                       │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ More Info     │ https://nodesecurity.io/advisories/566                                                                                 │
└───────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│               │ Prototype pollution attack                                                                                             │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Name          │ hoek                                                                                                                   │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ CVSS          │ 4 (Medium)                                                                                                             │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Installed     │ 2.16.3                                                                                                                 │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Vulnerable    │ <= 4.2.0 || >= 5.0.0 < 5.0.3                                                                                           │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Patched       │ > 4.2.0 < 5.0.0 || >= 5.0.3                                                                                            │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Path          │ hapi-auth-hawk@3.0.1 > hoek@2.16.3                                                                                     │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ More Info     │ https://nodesecurity.io/advisories/566                                                                                 │
└───────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│               │ Prototype pollution attack                                                                                             │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Name          │ hoek                                                                                                                   │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ CVSS          │ 4 (Medium)                                                                                                             │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Installed     │ 2.16.3                                                                                                                 │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Vulnerable    │ <= 4.2.0 || >= 5.0.0 < 5.0.3                                                                                           │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Patched       │ > 4.2.0 < 5.0.0 || >= 5.0.3                                                                                            │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Path          │ hapi-auth-hawk@3.0.1 > hawk@3.1.3 > hoek@2.16.3                                                                        │
├───────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ More Info     │ https://nodesecurity.io/advisories/566                                                                                 │
└───────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘


```

**Comment by mtharrison** [comment]
Can you use the latest hapi-auth-hawk@4.0.0? Or do you need hapi < 17 support?

**Comment by vladikoff** [comment]
Yeah, for now < 17

**Comment by mtharrison** [comment]
Can you please make a PR against the v3 branch and use `major.x.x` version specifiers for the hapi modules in `dependencies` please?

**Comment by hueniverse** [comment]
Still an issue?

**Comment by vladikoff** [comment]
Nevermind I just checked and v4 has the right fixes, thank you 👍

**Pull request closed without merging**